source/ota.c

```diff
@@ -478,7 +439,7 @@ static OtaErr_t setImageStateWithReason( OtaImageState_t stateToSet,
     /* Now update the image state and job status on service side. */
     otaAgent.imageState = state;
 
-    if( otaAgent.pOtaSingletonActiveJobName != NULL )
+    if( strlen( otaAgent.pActiveJobName ) > 0 )
```

Comment:
This can be user initiated setimage state by calling function OTA_SetImageState , so it can be called when no active job is available, so the expected behavior is only platform state is updated.